### DISCUSSION TITLE
Text Recognition Dataset - PaddleOCR

### DISCUSSION THREAD

**Discussion by Adhitya1996** [PaddlePaddle/PaddleOCR#13835] [Q&A]
Dear Paddle Community,

When I was currently preparing text recognition (already performed text detection with en_PP-OCRv3_det with ch_PP-OCRv3_det_student.yml), I noticed that most of the text recognition datasets primarly consists of only one word per image.

Example dataset of ICDAR can be downloaded from this link: https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.6/doc/doc_en/dataset/ocr_datasets_en.md and the training and testing images contain only one word per image.
Label can be accessed from here:https://[paddleocr.bj.bcebos.com/dataset/rec_gt_train.txt](https://paddleocr.bj.bcebos.com/dataset/rec_gt_train.txt)
It's format:
train/word_1.png	Genaxis Theatre
train/word_2.png	[06]
train/word_3.png	62-03
train/word_4.png	Carpark
.............

As you can see from above, they are single worded texts.

Query: 
1.What if the use case for me is to show a image where 10 clusters of words are present in it? 
How to train text recognition with en_PP-OCRv4_rec. How to format those images

2. Also, what is the difference between en_PP-OCRv4_rec and using an earlier version like en_PP-OCRv3_rec? 
Any Paddle documentation available to see the performance or some attributes of these models?

Thanks in advance!

Best Regards,
Adhitya

**Comment by GreatV** [comment]
1. please refer to: https://paddlepaddle.github.io/PaddleOCR/ppocr/model_train/training.html
2. for en_PP-OCRv4_rec please refer to: https://paddlepaddle.github.io/PaddleOCR/ppocr/model_list.html#22

**Reply by Adhitya1996** [reply]
Thank you so much for replying, it was kind of you to do so. 
I will look further into it regarding 2 but my 1st question is still not answered.

I do not have low training data or something. I was merely stating the fact that the text recognition of PaddleOCR data annotation receives only one word per image.

It's format:
train/word_1.png Genaxis Theatre
train/word_2.png [06]
train/word_3.png 62-03
train/word_4.png Carpark


Quer: How to train on a dataset where there 3 sentences with each sentence say having 5 words so 15 words in total. How to train text recognition with such datasets? Thanks, once again!

Best Regards,
Adhitya

**Reply by GreatV** [reply]
The dataset in the document is just an example. You can follow the instructions in the document to train on longer text directly without making any additional changes

**Reply by GreatV** [reply]
If the text is particularly long, you will need to modify the configuration file, such as changing `max_text_length`.

**Reply by Adhitya1996** [reply]
Hey, sorry for the late reply, your reply means a lot to me, above queries are cleared, thanks a ton!

I have two final theoretical queries to ask: 
1. Can you please explain what layers we are unfreezing during fine-tuning say for en_PP_OCRv3 text detection? 
Link for training: https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.3/doc/doc_ch/detection.md...
From this link for training, it is obvious that models runs thro architectures, transforms, backbones, neck, head. So essentially I think all layers are trained in training.
Query: What layers are frozen, unfrozen in fine-tuning of PaddleOCR text detection?

2. Can you give example of fine-tuning vs transfer learning specific to PaddleOCR text detection?

Thanks in advance!
